crates/extensions/c8y_mapper_ext/src/actor.rs

```diff
@@ -108,19 +109,25 @@ impl C8yMapperActor {
     }
 
     async fn process_mqtt_message(&mut self, message: MqttMessage) -> Result<(), RuntimeError> {
-        // to which entity is the message related
         if let Some(entity_id) = entity_store::entity_mqtt_id(&message.topic) {
-            // if message is registration message:
             if is_entity_register_message(&message) {
-                // update entity store with registration message
-                // TODO: emit error message on Err
-                self.entity_store
+                if let Err(e) = self
+                    .entity_store
                     .update(message.clone().try_into().unwrap())
-                    .unwrap();
+                {
+                    error!("Could not update device registration: {e}");
+                }
             } else {
                 // if device is unregistered register using auto-registration
                 if self.entity_store.get(entity_id).is_none() {
-                    let register_messages = self.auto_register_entity(entity_id);
+                    let register_messages = match self.auto_register_entity(entity_id) {
+                        Ok(register_messages) => register_messages,
+                        Err(e) => {
+                            error!("Could not update device registration: {e}");
```

Comment:
I feel it's better to write this error message to the `error` topic. @didier-wenzek can clarify this.

Reply:
To use or not the error topic is a broader topic that needs to be discussed with @reubenmiller .

Thin-edge emits error messages on too many channels. This is puzzling for the users (what streams  do I have to watch?) and the developers (where do I have to log this error?).

Ideally sending errors over MQTT should be done behind the scene by the `error!()` macro.